### PR TITLE
Ensure all WSGITransport environs have a SERVER_PROTOCOL

### PR DESCRIPTION
**Pull request by lazorchakp**
# Summary

- I noticed that WSGI environs sent to applications with `httpx.Client(app=app)` did not contain a SERVER_PROTOCOL
- [PEP-3333](https://peps.python.org/pep-3333/#environ-variables) indicates that this key must be present in the WSGI environ, and popular WSGI frameworks (such as [webob](https://github.com/Pylons/webob)) assume it will be as well
- It looks like httpx defaults to using `HTTP/1.1` for a nearly identical case in the ASGITransport, so I did the same here
- I'm more than happy to discuss any of this further!

# Checklist

- [x] I understand that this PR may be closed in case there was no previous discussion. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
  - _to my knowledge, this does not require a documentation update_

### CODE REVIEW

**Review by madkinsz — APPROVED** [review]
LGTM! Thanks for contributing.

Another maintainer may weigh in but it seems unlikely that this was intentionally excluded.

**Comment by lazorchakp** [comment]
Great, thanks! I'll give it a day or so before merging to see if anyone has other thoughts

**Comment by tomchristie** [comment]
valid.

**Comment by tomchristie** [comment]
Process improvements we can make here...

* @tomchristie - Thanks for the quick merge, though you don't need to always play that role. If you could allow other maintainers more space here too that'd probs be good for the team as a whole.
* @encode/maintainers - We ought to start making sure we include `CHANGELOG` changes as part of functional or bugfix PRs. That'll help ensure that we're properly describing the change from a user perspective, and will also make the release process simpler and less error-prone. We should probably update [the pull request template](https://github.com/encode/httpx/blob/master/.github/PULL_REQUEST_TEMPLATE.md) to include this as an item?
* @lazorchakp would you consider opening a PR to add this to the `CHANGELOG` under a new "development" heading?

**Comment by Kludex** [comment]
Yes. 👍

**Pull request merged**
